tools/code_policy_check.py

```
@@ -0,0 +1,262 @@
+
+
+# Register default per-file checks (headers first, then std:: scan).
+PER_FILE_CHECKS = [
+    _check_headers,
+    _check_std_usage,
+]
+
+
+def check_file(path, check_std_usage, header_mode):
+    try:
+        with open(path, "r", encoding="utf-8", errors="ignore") as handle:
```

Ruff (0.14.6)

Unnecessary mode argument

Remove mode argument

(UP015)